xbmc/addons/interfaces/GUI/ListItem.cpp

```diff
@@ -87,5 +112,236 @@ void Interface_GUIListItem::destroy(void* kodiBase, void* handle)
     delete item;
 }
 
+char* Interface_GUIListItem::get_label(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CFileItemPtr* item = static_cast<CFileItemPtr*>(handle);
+  if (!addon || !item)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, item, addon ? addon->ID().c_str() : "unknown");
+    return nullptr;
+  }
+
+  if (item->get() == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - empty list item called on addon '%s'", __FUNCTION__, addon->ID().c_str());
+    return nullptr;
+  }
+
+  char* ret;
+  Interface_GUIGeneral::lock();
+  ret = strdup(item->get()->GetLabel().c_str());
```

Comment:
Does not work and bring:
```
/home/alwin/Development/kodi/xbmc/addons/interfaces/GUI/ListItem.cpp: In static member function ‘static char* ADDON::Interface_GUIListItem::get_label(void*, void*)’:
/home/alwin/Development/kodi/xbmc/addons/interfaces/GUI/ListItem.cpp:134:23: error: ‘CFileItemPtr {aka class std::shared_ptr<CFileItem>}’ has no member named ‘GetLabel’
   ret = strdup(*item->GetLabel().c_str());
```
Thats why I leaved also as `if (item->get() == nullptr)` before.